### DISCUSSION TITLE
2022.2.3_2

### DISCUSSION THREAD

**Discussion by strogonoff** [ietf-tools/bibxml-service#108] [Announcements]
This release primarily deals with sourcing.

- Bibliographic data sources are now pluggable. Main Relaton sources [are registered in `main.sources`](https://github.com/ietf-ribose/bibxml-service/blob/8364c49ac02385405e98b2bb868041ba266cc235/main/sources.py#L97-L123), and xml2rfc [makes use](https://github.com/ietf-ribose/bibxml-service/blob/8364c49ac02385405e98b2bb868041ba266cc235/xml2rfc_compat/source.py#L75-L82) of it as well for indexing xml2rfc mirror repository for fallback purposes.
  - The associated refactor moved source-related functionality out from `management` module into either `sources` or `main`. Python module reference in documentation has been updated to reflect the current architecture.
- xml2rfc-style paths now [support fallback](https://github.com/ietf-ribose/bibxml-service/blob/8364c49ac02385405e98b2bb868041ba266cc235/xml2rfc_compat/urls.py#L132-L163) when up-to-date equivalent was not found for given anchor. This requires new `xml2rfc` source to be indexed (which can be done manually via management GUI, or via index API endpoint).

Other changes include:

- Container’s Dockerfile was updated to make better use of layer caching
- Monitoring services were split into auxiliary Compose configuration that can be used in conjunction with the main one via multiple `-f` options
- A possible crash when searching bibliographic data via API has been fixed
- API now conforms to the advertised spec better by returning JSON responses in case of bad request or server errors

<hr /><em>This discussion was created from the release <a href='https://github.com/ietf-ribose/bibxml-service/releases/tag/2022.2.3_2'>2022.2.3_2</a>.</em>

**Comment by strogonoff** [comment]
Note that after this update public development instance no longer seems to be running async tasks (see #106, #107). Compose configuration seems to work fine, so it doesn’t seem like it’s an app-level issue, but I’m not sure. cc @kwkwan @ronaldtse

**Reply by ronaldtse** [reply]
Maybe some issue in the deployment. Ping @kwkwan could you help check? Thanks.
